enabling 'filtered' on 'files' in assembly descriptor yields NullPointerException

example `assembly.xml` snippet

```xml
  <files>
    <file>
      <source>src/main/docker/app/startup.sh</source>
      <outputDirectory>/usr/local/bin</outputDirectory>
      <fileMode>0755</fileMode>
      <filtered>true</filtered>
    </file>
  </files>
```

relevant stacktrace

```
Caused by: java.lang.NullPointerException
	at org.apache.maven.plugin.assembly.format.ReaderFormatter.createReaderFilter(ReaderFormatter.java:81)
	at org.apache.maven.plugin.assembly.format.ReaderFormatter.access$000(ReaderFormatter.java:41)
	at org.apache.maven.plugin.assembly.format.ReaderFormatter$1.transform(ReaderFormatter.java:116)
	at org.apache.maven.plugin.assembly.archive.phase.FileItemAssemblyPhase$1.getInputStream(FileItemAssemblyPhase.java:130)
	at org.apache.maven.plugin.assembly.archive.phase.FileItemAssemblyPhase$Deferred.<init>(FileItemAssemblyPhase.java:166)
	at org.apache.maven.plugin.assembly.archive.phase.FileItemAssemblyPhase$1.<init>(FileItemAssemblyPhase.java:134)
	at org.apache.maven.plugin.assembly.archive.phase.FileItemAssemblyPhase.execute(FileItemAssemblyPhase.java:125)
	at org.apache.maven.plugin.assembly.archive.DefaultAssemblyArchiver.createArchive(DefaultAssemblyArchiver.java:162)
	at org.jolokia.docker.maven.assembly.DockerAssemblyManager.createArchiveFromAssembly(DockerAssemblyManager.java:126)
	at org.jolokia.docker.maven.assembly.DockerAssemblyManager.create(DockerAssemblyManager.java:55)
	at org.jolokia.docker.maven.BuildMojo.buildImage(BuildMojo.java:66)
	at org.jolokia.docker.maven.BuildMojo.executeInternal(BuildMojo.java:58)
	at org.jolokia.docker.maven.AbstractDockerMojo.execute(AbstractDockerMojo.java:136)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:133)
	... 20 more
```

this will require the `DockerAssemblyConfigurationSource` to return an implementation of `MavenReaderFilter`. i think this is a simple as returning an instance of `DefaultMavenReaderFilter`, i just nee to validate that.

--- a/README.md
+++ b/README.md
@@ -5,7 +5,7 @@
 [![Flattr](http://api.flattr.com/button/flattr-badge-large.png)](http://flattr.com/thing/73919/Jolokia-JMX-on-Capsaicin)
 
 This is a Maven plugin for managing Docker images and containers for your builds.
-The current version ist **0.11.0** and works with Maven 3.2.1 and Docker 1.3.0 or later.
+The current version ist **0.11.1** and works with Maven 3.2.1 and Docker 1.3.0 or later.
 
 The current Docker API version used is `v1.15` (which is the minimal required API version).
 See the **[User Manual](doc/manual.md)** for details on how to override this value for new

--- a/doc/changelog.md
+++ b/doc/changelog.md
@@ -1,5 +1,8 @@
 # ChangeLog
 
+* **0.11.2**
+  - fix NPE when filtering enabled during assembly creation (#82)
+
 * **0.11.1**
   - Add support for binding UDP ports (#83)
   - "Entrypoint" supports now arguments (#84)

--- a/pom.xml
+++ b/pom.xml
@@ -6,7 +6,7 @@
 
   <groupId>org.jolokia</groupId>
   <artifactId>docker-maven-plugin</artifactId>
-  <version>0.11.1-SNAPSHOT</version>
+  <version>0.11.1</version>
   <packaging>maven-plugin</packaging>
 
   <name>maven-docker-plugin</name>
@@ -264,7 +264,7 @@
   <scm>
     <connection>scm:git:git://github.com/rhuss/docker-maven-plugin.git</connection>
     <developerConnection>scm:git:ssh://git@github.com/rhuss/docker-maven-plugin.git</developerConnection>
-    <tag>HEAD</tag>
+    <tag>v0.11.1</tag>
     <url>git://github.com/rhuss/docker-maven-plugin.git</url>
   </scm>
 

--- a/pom.xml
+++ b/pom.xml
@@ -6,7 +6,7 @@
 
   <groupId>org.jolokia</groupId>
   <artifactId>docker-maven-plugin</artifactId>
-  <version>0.11.1</version>
+  <version>0.11.2-SNAPSHOT</version>
   <packaging>maven-plugin</packaging>
 
   <name>maven-docker-plugin</name>
@@ -264,7 +264,7 @@
   <scm>
     <connection>scm:git:git://github.com/rhuss/docker-maven-plugin.git</connection>
     <developerConnection>scm:git:ssh://git@github.com/rhuss/docker-maven-plugin.git</developerConnection>
-    <tag>v0.11.1</tag>
+    <tag>HEAD</tag>
     <url>git://github.com/rhuss/docker-maven-plugin.git</url>
   </scm>
 

--- a/samples/cargo-jolokia-demo/pom.xml
+++ b/samples/cargo-jolokia-demo/pom.xml
@@ -21,7 +21,7 @@
 
   <groupId>org.jolokia</groupId>
   <artifactId>docker-jolokia-demo</artifactId>
-  <version>0.11.0</version>
+  <version>0.11.1</version>
 
   <url>http://www.jolokia.org</url>
 

--- a/samples/data-jolokia-demo/pom.xml
+++ b/samples/data-jolokia-demo/pom.xml
@@ -21,7 +21,7 @@
 
   <groupId>org.jolokia</groupId>
   <artifactId>docker-jolokia-demo</artifactId>
-  <version>0.11.1-SNAPSHOT</version>
+  <version>0.11.1</version>
 
   <url>http://www.jolokia.org</url>
 

--- a/src/main/java/org/jolokia/docker/maven/BuildMojo.java
+++ b/src/main/java/org/jolokia/docker/maven/BuildMojo.java
@@ -6,6 +6,7 @@ import org.apache.maven.archiver.MavenArchiveConfiguration;
 import org.apache.maven.execution.MavenSession;
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.shared.filtering.MavenFileFilter;
+import org.apache.maven.shared.filtering.MavenReaderFilter;
 import org.jolokia.docker.maven.access.DockerAccess;
 import org.jolokia.docker.maven.access.DockerAccessException;
 import org.jolokia.docker.maven.assembly.DockerAssemblyManager;
@@ -38,6 +39,9 @@ public class BuildMojo extends AbstractDockerMojo {
     /** @component */
     private MavenFileFilter mavenFileFilter;
 
+    /** @component */
+    private MavenReaderFilter mavenFilterReader;
+    
     /** @component */
     private DockerAssemblyManager dockerAssemblyManager;
 
@@ -73,12 +77,15 @@ public class BuildMojo extends AbstractDockerMojo {
 
         checkImageWithAutoPull(dockerAccess, fromImage, new ImageName(fromImage).getRegistry());
 
-        MojoParameters params =  new MojoParameters(session, project, archive, mavenFileFilter, sourceDirectory, outputDirectory);
+        MojoParameters params = createMojoParameters();
         File dockerArchive = dockerAssemblyManager.create(params, imageConfig.getBuildConfiguration());
 
         dockerAccess.buildImage(imageName, dockerArchive);
         debug("Build successful!");
     }
 
-
+    private MojoParameters createMojoParameters() {
+        return new MojoParameters(session, project, archive, mavenFileFilter, mavenFilterReader,
+                sourceDirectory, outputDirectory);
+    }
 }

--- a/src/main/java/org/jolokia/docker/maven/assembly/DockerAssemblyConfigurationSource.java
+++ b/src/main/java/org/jolokia/docker/maven/assembly/DockerAssemblyConfigurationSource.java
@@ -222,7 +222,7 @@ public class DockerAssemblyConfigurationSource implements AssemblerConfiguration
 
     @Override
     public MavenReaderFilter getMavenReaderFilter() {
-        return null;
+        return params.getMavenFilterReader();
     }
 
     @Override

--- a/src/main/java/org/jolokia/docker/maven/util/MojoParameters.java
+++ b/src/main/java/org/jolokia/docker/maven/util/MojoParameters.java
@@ -4,6 +4,7 @@ import org.apache.maven.archiver.MavenArchiveConfiguration;
 import org.apache.maven.execution.MavenSession;
 import org.apache.maven.project.MavenProject;
 import org.apache.maven.shared.filtering.MavenFileFilter;
+import org.apache.maven.shared.filtering.MavenReaderFilter;
 
 /**
  * Helper class for encapsulating Mojo params which are not Plexus components
@@ -12,19 +13,21 @@ import org.apache.maven.shared.filtering.MavenFileFilter;
  * @since 09.05.14
  */
 public class MojoParameters {
-    private MavenArchiveConfiguration archive;
-    private MavenSession session;
-    private MavenFileFilter mavenFileFilter;
-    private MavenProject project;
+    private final MavenArchiveConfiguration archive;
+    private final MavenSession session;
+    private final MavenFileFilter mavenFileFilter;
+    private final MavenReaderFilter mavenFilterReader;
+    private final MavenProject project;
 
     private final String outputDirectory;
     private final String sourceDirectory;
 
     public MojoParameters(MavenSession session, MavenProject project, MavenArchiveConfiguration archive, MavenFileFilter mavenFileFilter,
-            String sourceDirectory, String outputDirectory) {
+            MavenReaderFilter mavenFilterReader, String sourceDirectory, String outputDirectory) {
         this.archive = archive;
         this.session = session;
         this.mavenFileFilter = mavenFileFilter;
+        this.mavenFilterReader = mavenFilterReader;
         this.project = project;
 
         this.sourceDirectory = sourceDirectory;
@@ -49,6 +52,10 @@ public class MojoParameters {
 
     public MavenFileFilter getMavenFileFilter() {
         return mavenFileFilter;
+    }
+    
+    public MavenReaderFilter getMavenFilterReader() {
+        return mavenFilterReader;
     }
 
     public MavenProject getProject() {

--- a/src/test/java/org/jolokia/docker/maven/assembly/DockerAssemblyConfigurationSourceTest.java
+++ b/src/test/java/org/jolokia/docker/maven/assembly/DockerAssemblyConfigurationSourceTest.java
@@ -38,13 +38,13 @@ public class DockerAssemblyConfigurationSourceTest {
     private MojoParameters buildParameters(String projectDir, String sourceDir, String outputDir) {
         MavenProject mavenProject = new MavenProject();
         mavenProject.setFile(new File(projectDir));
-        return new MojoParameters(null, mavenProject, null, null, sourceDir, outputDir);
+        return new MojoParameters(null, mavenProject, null, null, null, sourceDir, outputDir);
     }
 
     @Test
     public void testEmptyAssemblyConfig() {
         DockerAssemblyConfigurationSource source = new DockerAssemblyConfigurationSource(
-                new MojoParameters(null, null, null, null, "/src/docker", "/output/docker"),
+                new MojoParameters(null, null, null, null, null, "/src/docker", "/output/docker"),
                 null
         );
         assertEquals(0,source.getDescriptors().length);

